### PR TITLE
Add vshn postgres

### PR DESCRIPTION
**Pull request by zugao**
* This PR adds crossplane support for stackgres operator.
* Added composite and composition
* Tested in my local cluster

## Checklist

- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: component/vshn_postgres.jsonnet [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+  'POSTGRESQL_DB',
+  'POSTGRESQL_HOST',
+  'POSTGRESQL_PORT',
+  'POSTGRESQL_USER',
+  'POSTGRESQL_PASSWORD',
+  'ca.crt',
+];
+
+local xrd = xrds.XRDFromCRD(
+  'xvshnpostgresqls.appcat.vshn.io',
```
Comment (glrf) [inline component/vshn_postgres.jsonnet:31]: Did we define if the api group should be `appcat.vshn.io` or `vshn.appcat.vshn.io`?

File: Makefile.vars.mk [hunk 2]
```diff
@@ -46,4 +46,4 @@ KUBENT_IMAGE    ?= ghcr.io/doitintl/kube-no-trouble:latest
 KUBENT_DOCKER   ?= $(DOCKER_CMD) $(DOCKER_ARGS) $(root_volume) --entrypoint=/app/kubent $(KUBENT_IMAGE)
 
 instance ?= defaults
-test_instances = tests/defaults.yml tests/exoscale.yml tests/cloudscale.yml tests/openshift.yml
+test_instances = tests/defaults.yml tests/exoscale.yml tests/cloudscale.yml tests/openshift.yml tests/vshn.yml
```
Comment (glrf) [inline Makefile.vars.mk:49]: The new test case is missing in the github workflow and in the `.cruft.json`. 

We're now onboarded to the component template, you can use `commodore component update -t vshn .` to add a new testcase called `vshn`

**Review by glrf — COMMENTED** [review]
Looks pretty good overall, I like the new helper functions!

There are just a few smaller things and nitpicks

File: component/vshn_postgres.jsonnet [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+                      spec+: {
+                        forProvider+: {
+                          manifest+: {
+                            metadata: {
+                              name: '',
+                            },
+                          },
+                        },
+                      },
+                    };
```
[the remark this answers is not in the document]
Comment (zugao) [reply component/vshn_postgres.jsonnet:50]: no problem

File: component/vshn_postgres.jsonnet (hunk 1 above)
Comment (zugao) [reply component/vshn_postgres.jsonnet]: It's `vshn.appcat.vshn.io`. Updated the code.

**Review by glrf — APPROVED** [review]
LGTM (haven't tested it locally though)

File: component/vshn_postgres.jsonnet [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                    {
+                      spec+: {
+                        forProvider+: {
+                          manifest+: {
+                            metadata: {
+                              name: '',
+                              namespace: '',
+                            },
+                            spec: {
+                              instances: 1,
```
Comment (susana-garcia) [inline component/vshn_postgres.jsonnet:98]: is there a reason why the value of instances is 1? shouldn't it be patched?

**Review by susana-garcia — APPROVED** [review]
LGTM, haven't test it locally and I just have a question.

File: component/vshn_postgres.jsonnet (hunk 4 above)
Comment (zugao) [reply component/vshn_postgres.jsonnet:98]: We decided to hard code it to 1 for now.

**Comment by Kidswiss** [comment]
One point I'm not sure if it was looked at:

> The customer should not have any access to the namespace kubernetes resources

I'm fine with moving that to a new story if it's too much for the context of this story.

**Pull request merged**